roles/edpm_kernel/tasks/kernelargs.yml

```diff
@@ -100,13 +100,25 @@
         - /boot/efi/EFI/fedora
 
     - name: Generate grub config
-      ansible.builtin.command: "grub2-mkconfig -o /boot/grub2/grub.cfg"
+      ansible.builtin.command: >-
+        grub2-mkconfig -o /boot/grub2/grub.cfg
+        {{ '--update-bls-cmdline'
+        if ansible_distribution == 'RedHat'
```

Comment:
so the main open is do we want to also do this if centos 9 stream
or only apply this for downstream/rhel builds

for now we could be conservative and only do it for rhel but my inclination is both.

lest see what others think.

in terms of testing it would be nice to update molecule to have a test for this but that could be done later.
in general i would personally prefer to see this executed in at least one fo our jobs before this merges but lets see what others think.

Reply:
so the issue is that the existing test is configured to use the local connection and effectivly the delegated driver

https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/roles/edpm_kernel/molecule/kernelargs-update/molecule.yml#L3-L10

which mean that test is running on the os provide by the zuul VM as defined by the label in the relevant nodeset

so its currently running on centos and as a result this wont run.

the current test also only assets the content of /etc/default/grub

https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/roles/edpm_kernel/molecule/kernelargs-update/verify.yml#L18-L46

so its not actually testing the current functionally properly
(its not asserting that /boot/grub2/grub.cfg has the expected output)
 and has not been extended for the bsl functionality

we should update the test on rhel to assert that 
the kernel args are in the output of  "sudo grubby --info ALL" i think.

with the current check to have that run in ci upstream we would have to update the nodeset for this molecule job too to use rhel 9.3+

Reply:
it looks like there is a cloud-rhel-9-3 label https://review.rdoproject.org/zuul/labels

we do not currently have a nodeset we can directly use but we do have the ability to use labels with rhel

https://github.com/openstack-k8s-operators/ci-framework/blob/66277da980a5664413438f32c522739c6ede987a/zuul.d/nodeset.yaml#L46-L63

for example the centos-9-rhel-9-2-crc-extracted-2.30-3xl nodeset.

so what is probably reasonable here would be to either add centos support and use the existing job or file an issue to add rhel 9 based test coverage.

in either case it would be good to consider if the molecule tests could be enhanced to  assert the bsl entreise are configured properly since /boot/grub2/grub.cfg will be ingored on rhel 9.3+